napari/layers/__init__.py

```diff
@@ -31,6 +32,7 @@
     'Layer',
     'Points',
     'Shapes',
+    'SpatialInformation',
```

Comment:
I'm worried about exposing a new name in the public API (both in `napari.layers.utils` and in `napari.layers`) until there's more discussion about it and until it's actually used by napari layers (so users need a public way to deal with it).

For now, I suggest this in privately name module/package.

Reply:
My idea is to remove spacing etc for layers itself (also from the constructor) To reduce the number of arguments. So in future I would like to layer creator provide this information as a single object (or dict).

Reply:
Understood.

First, I though this PR would just merge this structure in without using it (in which case a private module is fine).

But now I understand that you plan to use it, then agreed we should make this public. Now just need to find a name people can agree on!